### PR TITLE
[dv,usbdev] Improve testing of suspend transitions

### PR DESCRIPTION
**Pull request by alees24**
Some of the link transitions were previously not observed because (i) the driver/sequences typically do not model the USB by periodically sending the SOF packets when the bus is Idle. (ii) the Powered link state is transient and difficult to observe because the DP/DN both low state before VBUS connection acts as a Bus Reset. It therefore tends to transition quickly to ActiveNoSOF. Update the BFM to reflect the Powered -> PoweredSuspended transition.

Randomization is now required for the `usbdev_rx_full` sequence to be useful, since its scope has been extended.

### CODE REVIEW

File: hw/ip/usbdev/dv/usbdev_sim_cfg.hjson [hunk 1]
```diff
@@ -450,8 +450,6 @@
     {
       name: usbdev_rx_full
       uvm_test_seq: usbdev_rx_full_vseq
-      // No need for randomization.
-      reseed: 1
```
Comment (andreaskurth) [inline hw/ip/usbdev/dv/usbdev_sim_cfg.hjson:454]: Is this change intended to be part of this PR?

**Review by andreaskurth — APPROVED** [review]
This looks sensible to me; just one question

File: hw/ip/usbdev/dv/usbdev_sim_cfg.hjson (hunk 1 above)
Comment (alees24) [reply hw/ip/usbdev/dv/usbdev_sim_cfg.hjson:454]: Yes, as per the final comment. Perhaps you'd prefer it as a separate commit or PR? It should have been changed when that sequence was changed a couple of days ago.

Comment (andreaskurth) [reply hw/ip/usbdev/dv/usbdev_sim_cfg.hjson:454]: Ah indeed, sorry. Yes, I think a separate commit would be preferable. I'll leave the decision about keeping it in this PR or creating a new one to you

Comment (alees24) [reply hw/ip/usbdev/dv/usbdev_sim_cfg.hjson:454]: Okay, separated. I'll just let it run the basic checks again, thanks.
